utils/node_local_test.py

```diff
@@ -1662,18 +1662,290 @@ def test_pydaos_kv(server, conf):
+            # If a fault wasn't injected then check output is as expected.
+            # It's not possible to log these as warnings, because there is
+            # no src line to log them against, so simply assert.
+            assert self.returncode == 0
+            assert self.stderr == b''
+            if self.aft.expected_stdout is not None:
+                assert self.stdout == self.aft.expected_stdout
+            self.fault_injected = False
+        if self.vh:
+            self.vh.convert_xml()
+        if not self.fault_injected:
+            return
+        if not self.aft.check_stderr:
+            return
+
+        if self.returncode == 0:
+            if self.stdout != self.aft.expected_stdout:
+                self.aft.conf.wf.add(self.fi_loc,
+                                     'NORMAL',
+                                     "Incorrect stdout '{}'".format(
+                                         self.stdout),
+                                     mtype='Out of memory caused zero exit '
+                                     'code with incorrect output')
+
+        stderr = self.stderr.decode('utf-8').rstrip()
+        if not stderr.endswith("Out of memory (-1009)") and \
+           'error parsing command line arguments' not in stderr and \
+           self.stdout != self.aft.expected_stdout:
+            if self.stdout != b'':
+                print(self.aft.expected_stdout)
+                print()
+                print(self.stdout)
+                print()
+            self.aft.conf.wf.add(self.fi_loc,
+                                 'NORMAL',
+                                 "Incorrect stderr '{}'".format(
+                                     stderr),
+                                 mtype='Out of memory not reported '
+                                 'correctly via stderr')
+
+class AllocFailTest():
+    """Class to describe fault injection command"""
+
+    def __init__(self, conf, cmd):
+        self.conf = conf
+        self.cmd = cmd
+        self.prefix = True
+        self.check_stderr = False
+        self.expected_stdout = None
+        self.use_il = False
+
+    def launch(self):
+        """Run all tests for this command"""
+
+        def _prep(self):
+            rc = self._run_cmd(None)
+            rc.wait()
+            self.expected_stdout = rc.stdout
+            assert not rc.fault_injected
+
+        # Prep what the expected stdout is by running once without faults
+        # enabled.
+        _prep(self)
+
+        print('Expected stdout is')
+        print(self.expected_stdout)
+
+        num_cores = len(os.sched_getaffinity(0))
+
+        if num_cores < 20:
+            max_child = 1
+        else:
+            max_child = int(num_cores / 4 * 3)
+
+        active = []
+        fid = 1
+        max_count = 0
+        finished = False
+
+        # List of fids to re-run under valgrind.
+        to_rerun = []
+
+        fatal_errors = False
+
+        while not finished or active:
```

Comment:
I personally think this is easier to read if you add parenthesis around "not finished" so one doesn't need to remember order of operations rules

Reply:
I think pylint would warn if I tried that, but I'll address if I need to revise this.